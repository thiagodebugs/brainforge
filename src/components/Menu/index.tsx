"use client";
import Link from "next/link";
import Icon from "@mdi/react";
import style from "./style.module.css";
import {
  mdiHome,
  mdiRocketLaunch,
  mdiAccountSearch,
  mdiCogs,
  mdiFinance,
} from "@mdi/js";
import { useState } from "react";

export default function Menu() {
  const [active, setActive] = useState(1);

  return (
    <>
      <div className={style.menu}>
        <ul>
          <Link
            href="#"
            className={`${style.btnMenu} ${active === 1 && style.active}`}
            onClick={() => setActive(1)}
          >
            <li>
              <Icon path={mdiHome} size={0.8} />
              <span>In√≠cio</span>
            </li>
          </Link>
          <Link
            href="#"
            className={`${style.btnMenu} ${active === 2 && style.active}`}
            onClick={() => setActive(2)}
          >
            <li>
              <Icon path={mdiRocketLaunch} size={0.8} />
              <span>Plano de Carreira</span>
            </li>
          </Link>
          <Link
            href="#"
            className={`${style.btnMenu} ${active === 3 && style.active}`}
            onClick={() => setActive(3)}
          >
            <li>
              <Icon path={mdiAccountSearch} size={0.8} />
              <span>Vagas</span>
            </li>
          </Link>
          <Link
            href="#"
            className={`${style.btnMenu} ${active === 4 && style.active}`}
            onClick={() => setActive(4)}
          >
            <li>
              <Icon path={mdiCogs} size={0.8} />
              <span>Processos</span>
            </li>
          </Link>
          <Link
            href="#"
            className={`${style.btnMenu} ${active === 5 && style.active}`}
            onClick={() => setActive(5)}
          >
            <li>
              <Icon path={mdiFinance} size={0.8} />
              <span>PowerBI</span>
            </li>
          </Link>
        </ul>
      </div>
    </>
  );
}
