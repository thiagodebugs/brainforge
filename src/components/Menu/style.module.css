.menu {
  @apply fixed bottom-2 px-2 xl:px-0 z-50 left-0 right-0 xl:relative w-full;
}

.menu ul {
  @apply font-extralight text-sm  grid grid-cols-5 xl:grid-cols-1 gap-8 xl:gap-2 max-w-md xl:max-w-none mx-auto bg-[#131313] xl:bg-transparent rounded-full px-6 py-2 xl:px-0 xl:py-0;
}

.btnMenu {
  @apply border-0 xl:border border-zinc-600 rounded-full py-1 px-3 flex justify-center items-center xl:justify-start;
  transition: all 0.3s ease-in-out;
}

.btnMenu li {
  @apply flex justify-start items-center space-x-2;
}

.btnMenu li span {
  @apply hidden xl:block;
}

.btnMenu li svg {
  @apply w-6 h-6 xl:w-5 xl:h-5 !important;
}

.active {
  @apply font-bold bg-[#26c666] text-white xl:text-[#26c666] xl:font-normal xl:bg-transparent border-[#26c666] border-opacity-60;
}

.active svg {
  @apply xl:text-[#26c666];
}

.btnMenu:hover {
  @apply font-bold bg-[#26c666] text-white;
}

.btnMenu:hover svg {
  @apply text-white;
}
