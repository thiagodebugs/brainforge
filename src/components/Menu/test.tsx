import { render } from "@testing-library/react";
import Menu from "./";

describe("Menu", () => {
  it("should render the menu", () => {
    const { container } = render(<Menu />);

    expect(container).toMatchSnapshot();
  });

  it("click buttons", () => {
    const { getByText } = render(<Menu />);

    expect(getByText("In√≠cio")).toBeInTheDocument();
    expect(getByText("Plano de Carreira")).toBeInTheDocument();
    expect(getByText("Vagas")).toBeInTheDocument();
    expect(getByText("Processos")).toBeInTheDocument();
    expect(getByText("PowerBI")).toBeInTheDocument();
  });
});
