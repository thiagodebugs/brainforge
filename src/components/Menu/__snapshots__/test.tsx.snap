// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render the menu 1`] = `
<div>
  <div
    class="menu"
  >
    <ul>
      <a
        class="btnMenu active"
        href="#"
      >
        <li>
          <svg
            role="presentation"
            style="width: 1.2000000000000002rem; height: 1.2000000000000002rem;"
            viewBox="0 0 24 24"
          >
            <path
              d="M10,20V14H14V20H19V12H22L12,3L2,12H5V20H10Z"
              style="fill: currentColor;"
            />
          </svg>
          <span>
            In√≠cio
          </span>
        </li>
      </a>
      <a
        class="btnMenu false"
        href="#"
      >
        <li>
          <svg
            role="presentation"
            style="width: 1.2000000000000002rem; height: 1.2000000000000002rem;"
            viewBox="0 0 24 24"
          >
            <path
              d="M13.13 22.19L11.5 18.36C13.07 17.78 14.54 17 15.9 16.09L13.13 22.19M5.64 12.5L1.81 10.87L7.91 8.1C7 9.46 6.22 10.93 5.64 12.5M21.61 2.39C21.61 2.39 16.66 .269 11 5.93C8.81 8.12 7.5 10.53 6.65 12.64C6.37 13.39 6.56 14.21 7.11 14.77L9.24 16.89C9.79 17.45 10.61 17.63 11.36 17.35C13.5 16.53 15.88 15.19 18.07 13C23.73 7.34 21.61 2.39 21.61 2.39M14.54 9.46C13.76 8.68 13.76 7.41 14.54 6.63S16.59 5.85 17.37 6.63C18.14 7.41 18.15 8.68 17.37 9.46C16.59 10.24 15.32 10.24 14.54 9.46M8.88 16.53L7.47 15.12L8.88 16.53M6.24 22L9.88 18.36C9.54 18.27 9.21 18.12 8.91 17.91L4.83 22H6.24M2 22H3.41L8.18 17.24L6.76 15.83L2 20.59V22M2 19.17L6.09 15.09C5.88 14.79 5.73 14.47 5.64 14.12L2 17.76V19.17Z"
              style="fill: currentColor;"
            />
          </svg>
          <span>
            Plano de Carreira
          </span>
        </li>
      </a>
      <a
        class="btnMenu false"
        href="#"
      >
        <li>
          <svg
            role="presentation"
            style="width: 1.2000000000000002rem; height: 1.2000000000000002rem;"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5,12C18,12 20,14 20,16.5C20,17.38 19.75,18.21 19.31,18.9L22.39,22L21,23.39L17.88,20.32C17.19,20.75 16.37,21 15.5,21C13,21 11,19 11,16.5C11,14 13,12 15.5,12M15.5,14A2.5,2.5 0 0,0 13,16.5A2.5,2.5 0 0,0 15.5,19A2.5,2.5 0 0,0 18,16.5A2.5,2.5 0 0,0 15.5,14M10,4A4,4 0 0,1 14,8C14,8.91 13.69,9.75 13.18,10.43C12.32,10.75 11.55,11.26 10.91,11.9L10,12A4,4 0 0,1 6,8A4,4 0 0,1 10,4M2,20V18C2,15.88 5.31,14.14 9.5,14C9.18,14.78 9,15.62 9,16.5C9,17.79 9.38,19 10,20H2Z"
              style="fill: currentColor;"
            />
          </svg>
          <span>
            Vagas
          </span>
        </li>
      </a>
      <a
        class="btnMenu false"
        href="#"
      >
        <li>
          <svg
            role="presentation"
            style="width: 1.2000000000000002rem; height: 1.2000000000000002rem;"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.9,18.45C17.25,18.45 18.35,17.35 18.35,16C18.35,14.65 17.25,13.55 15.9,13.55C14.54,13.55 13.45,14.65 13.45,16C13.45,17.35 14.54,18.45 15.9,18.45M21.1,16.68L22.58,17.84C22.71,17.95 22.75,18.13 22.66,18.29L21.26,20.71C21.17,20.86 21,20.92 20.83,20.86L19.09,20.16C18.73,20.44 18.33,20.67 17.91,20.85L17.64,22.7C17.62,22.87 17.47,23 17.3,23H14.5C14.32,23 14.18,22.87 14.15,22.7L13.89,20.85C13.46,20.67 13.07,20.44 12.71,20.16L10.96,20.86C10.81,20.92 10.62,20.86 10.54,20.71L9.14,18.29C9.05,18.13 9.09,17.95 9.22,17.84L10.7,16.68L10.65,16L10.7,15.31L9.22,14.16C9.09,14.05 9.05,13.86 9.14,13.71L10.54,11.29C10.62,11.13 10.81,11.07 10.96,11.13L12.71,11.84C13.07,11.56 13.46,11.32 13.89,11.15L14.15,9.29C14.18,9.13 14.32,9 14.5,9H17.3C17.47,9 17.62,9.13 17.64,9.29L17.91,11.15C18.33,11.32 18.73,11.56 19.09,11.84L20.83,11.13C21,11.07 21.17,11.13 21.26,11.29L22.66,13.71C22.75,13.86 22.71,14.05 22.58,14.16L21.1,15.31L21.15,16L21.1,16.68M6.69,8.07C7.56,8.07 8.26,7.37 8.26,6.5C8.26,5.63 7.56,4.92 6.69,4.92A1.58,1.58 0 0,0 5.11,6.5C5.11,7.37 5.82,8.07 6.69,8.07M10.03,6.94L11,7.68C11.07,7.75 11.09,7.87 11.03,7.97L10.13,9.53C10.08,9.63 9.96,9.67 9.86,9.63L8.74,9.18L8,9.62L7.81,10.81C7.79,10.92 7.7,11 7.59,11H5.79C5.67,11 5.58,10.92 5.56,10.81L5.4,9.62L4.64,9.18L3.5,9.63C3.41,9.67 3.3,9.63 3.24,9.53L2.34,7.97C2.28,7.87 2.31,7.75 2.39,7.68L3.34,6.94L3.31,6.5L3.34,6.06L2.39,5.32C2.31,5.25 2.28,5.13 2.34,5.03L3.24,3.47C3.3,3.37 3.41,3.33 3.5,3.37L4.63,3.82L5.4,3.38L5.56,2.19C5.58,2.08 5.67,2 5.79,2H7.59C7.7,2 7.79,2.08 7.81,2.19L8,3.38L8.74,3.82L9.86,3.37C9.96,3.33 10.08,3.37 10.13,3.47L11.03,5.03C11.09,5.13 11.07,5.25 11,5.32L10.03,6.06L10.06,6.5L10.03,6.94Z"
              style="fill: currentColor;"
            />
          </svg>
          <span>
            Processos
          </span>
        </li>
      </a>
      <a
        class="btnMenu false"
        href="#"
      >
        <li>
          <svg
            role="presentation"
            style="width: 1.2000000000000002rem; height: 1.2000000000000002rem;"
            viewBox="0 0 24 24"
          >
            <path
              d="M6,16.5L3,19.44V11H6M11,14.66L9.43,13.32L8,14.64V7H11M16,13L13,16V3H16M18.81,12.81L17,11H22V16L20.21,14.21L13,21.36L9.53,18.34L5.75,22H3L9.47,15.66L13,18.64"
              style="fill: currentColor;"
            />
          </svg>
          <span>
            PowerBI
          </span>
        </li>
      </a>
    </ul>
  </div>
</div>
`;
